winsup/utils/getprocaddr.c

```diff
@@ -231,9 +238,29 @@ main (int argc, char **argv)
       return 1;
     }
 
-  /* This is needed for both thread to inject
-   * and GenerateConsoleCtrlEvent.
-   * Gives 1 sec time. */
-  Sleep (1000);
+  CtrlEvent = CreateEvent (NULL,              // default security attributes
+                           TRUE,              // manual-reset event
+                           FALSE,             // initial state is nonsignaled
+                           TEXT ("CtrlEvent") // object name
+  );
+  if (CtrlEvent == NULL)
+    {
+      output (1, "CreateEvent failed (%d)\n", GetLastError ());
+      return 1;
+    }
+
+  DWORD WaitResult = WaitForSingleObject (CtrlEvent, 10000 /* 10 seconds*/);
+
+  switch (WaitResult)
+    {
+    // All thread objects were signaled
+    case WAIT_OBJECT_0:
+      break;
+
+    // An error occurred
+    default:
+      output (1, "WaitForSingleObject failed (%d)\n", GetLastError ());
+      return 1;
+    }
```

Comment:
Wouldn't this be much easier to read as `if (WaitResult != WAIT_OBJECT_0) { fprintf(stderr, ...); return 1; }`?